sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java

```diff
@@ -77,10 +77,30 @@ public MarketplacePresenter(@NonNull final IMarketplaceView view, @NonNull final
 	@Override
 	public void onAttach(IMarketplaceView view) {
 		super.onAttach(view);
+		eventLogger.send(MarketplacePageViewed.create());
+	}
+
+	@Override
+	public void onDetach() {
+		super.onDetach();
+		navigator = null;
```

Comment:
you shouldn't set navigator = null -> the scope of MarketplacePresenter.java is handled by EcosystemActivity navigator shouldn't be set to null if MarketplacePresenter is still usable from EcosystemActivity

Reply:
it's not true, because the marketplacePresenter can hold an instance for the activity.
So when the activity is destroyed, (going back from Marketplace), the presenter still hold the activity instance and leak it. I can see it LeakCanary